File: src/main/java/com/github/spb/tget/demo/util/CommonUtils.java
package com.github.spb.tget.demo.util;

import org.apache.commons.lang3.StringUtils;

public class CommonUtils {

    public static String getSubstringBetweenOrToEnd(String source, String from, String to) {
        if (!source.contains(from)) {
            return "";
        }

        String substringFrom = StringUtils.substringAfter(source, from);
        if (substringFrom.contains(to)) {
            return StringUtils.substringBefore(substringFrom, to);
        }

        return substringFrom;
    }
}

File: Service/src/main/java/com/github/spb/tget/demo/util/CommonUtils.kt
package com.github.spb.tget.demo.util

import org.apache.commons.lang3.StringUtils

fun String.getSubstringBetweenOrToEnd(from: String, to: String): String {
    if (!this.contains(from)) {
        return ""
    }

    val substringFrom = StringUtils.substringAfter(this, from)
    return if (substringFrom.contains(to)) {
        StringUtils.substringBefore(substringFrom, to)
    } else substringFrom
}